independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java

```diff
@@ -179,10 +173,12 @@ public void reset(Map<String, byte[]> resources,
                 //ignore
             }
         }
-        for (ClassPathElement i : elements) {
-            ClassPathResource res = i.getResource(nm);
-            if (res != null) {
-                resources.add(res.getUrl());
+        //TODO: in theory resources could have been added in dev mode
+        //but I don't thing this really matters for this code path
+        ClassPathElement[] providers = state.loadableResources.get(name);
+        if (providers != null) {
+            for (ClassPathElement element : providers) {
+                resources.add(element.getResource(nm).getUrl());
```

Comment:
I tried out my application with this pr. I get an NPE at this line now.
Drools 6.5.0.Final loads some resources.

Value of `nm`: `/META-INF/drools.default.packagebuilder.conf`
Value of `name`: `META-INF/drools.default.packagebuilder.conf`

element.getResource(nm) returns null.
element.getResource(name) returns a value.

Reply:
@stuartwdouglas I let you check that comment.

Reply:
There was a typo, some places were using the original name that did not have the / removed